### ISSUE TITLE
[ramuda] Add wiring for cloudfront to use with lambda@edge

### ISSUE THREAD

**Issue by c0un7-z3r0** [glomex/gcdt#381]
To use lambda@edge we need to be able to add a trigger for cloudfront to the lambda.
[Docu for lambda@edge](http://docs.aws.amazon.com/lambda/latest/dg/lambda-edge.html)

Some things that have to be considered are that the trigger can only be applied to a replica version of that lambda in us-east-1. After the trigger has been applied that lambda replica will be distributed to the edge locations of the cloudfront.

**Comment by markfink** [comment]
confirmed with @olysyuk via slack that this currently the most important enhancement for gcdt. this feature will be implemented in ramuda. currently lamgda@edge is deployed via us-east-1 also ramudas ACTIVE alias does not work with lambda@edge.

**Comment by pmdroid** [comment]
@okinash @markfink the support of lambda edge would be quite interesting for us

**Comment by markfink** [comment]
I looked into, built a test setup and now I am working on the implementation of the feature. I am able to share some insights with you now. 

**limitations**

* can not use artifact bucket for upload
* max memory use 128 MB
* max timeout 3 sec
* lambda@edge does not implement alias or $LATEST so we use the version of the latest published
  version for wiring
* unwire removes the lambda trigger for the configured CloudFront distribution (regardless if the lambda function is the same or not)

**ramuda config**

``` js
...
"events": [{
    "event_source": {
        "arn": "arn:aws:cloudfront::420189626185:distribution/E1V934UN4EJGJA",
        "cache_behavior": "*",
        "cloudfront_event": "origin-request"
    }
}]
```

In case you want sth. else please let me know asap.

**Comment by c0un7-z3r0** [comment]
The max timeout is 3 sec only for CloudFront origin request and origin response events and for CloudFront viewer request events and viewer response events it is 1 second.

**Comment by olysyuk** [comment]
I agree with proposed solution

**Comment by markfink** [comment]
@Dgadavin gave his feedback 'ok' via slack on 08.11.2017 15:23

**Comment by markfink** [comment]
just released this with version 0.1.439. @Dgadavin is testing it on his stack now. Please let me know how it goes. thank you

**Comment by Dgadavin** [comment]
I've tested lambda@edge wiring. Everything looks good. We can close the issue.
